### ISSUE TITLE
[Bug]: language-* class of inline code in markdown always outputs as "language-plaintext"

### ISSUE THREAD

**Issue by juyuwol** [jekyll/jekyll#8990]
### Operating System

Windows 10

### Ruby Version

ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x64-mingw32]

### Jekyll Version

jekyll 4.2.2

### Expected Behavior

When "\`\<html\>\`{:.language-html}" was inputted, it should be "\<code class="language-html"\> ... code ... \</code\>".

### Current Behavior

language-* class becomes always "language-plaintext". But syntax highlighting works correctly according to the language class on input.

**Comment by juyuwol** [comment]
For reasons I don't understand and therefore cannot explain, commenting out [this line](https://github.com/jekyll/jekyll/blob/3f46f02108048f2151724220d0037cbce09f483e/lib/jekyll/converters/markdown/kramdown_parser.rb#L102) solves the problem.

jekyll/lib/jekyll/converters/markdown/kramdown_parser.rb:

``` ruby
        def setup
          @config["syntax_highlighter"] ||= highlighter
          @config["syntax_highlighter_opts"] ||= {}
          # @config["syntax_highlighter_opts"]["default_lang"] ||= "plaintext"
          @config["syntax_highlighter_opts"]["guess_lang"] = @config["guess_lang"]
          @config["coderay"] ||= {} # XXX: Legacy.
          modernize_coderay_config
        end
```

I hope someone can test this out.

**Comment by ashmaroli** [comment]
@juyuwol This is not a bug but intended by design. See: #8035

**Comment by juyuwol** [comment]
@ashmaroli I'm not talking about plaintext being the default language.

[The Kramdown specification states that you can specify a language as IAL in code span](https://kramdown.gettalong.org/syntax.html#code-spans). If the output is plaintext instead of the language specified by the class name, it must be a bug, right?

**Comment by juyuwol** [comment]
For clarity, I'll explain the detailed behavior.

***

**Input (Markdown)**:

\`&lt;html&gt;\`{:.language-html}

**Expected Output (HTML)**:

&lt;code class="language-html highlighter-rouge"&gt;&lt;span class="nt"&gt;&amp;lt;html&amp;gt;&lt;/span&gt;&lt;/code&gt;

**Actual Output (HTML)**:

&lt;code class="language-plaintext highlighter-rouge"&gt;&lt;span class="nt"&gt;&amp;lt;html&amp;gt;&lt;/span&gt;&lt;/code&gt;

***

I'm pretty sure this is not the intended behavior.

**Comment by ashmaroli** [comment]
Okay @juyuwol, I understand the issue better now.
Somehow, I feel there is a *discrepancy* at kramdown's end as well. Try the following snippet:

~~~
This is a Ruby code fragment `x = Class.new`{:.language-ruby}

```
def what?
  42
end
```
{: .language-ruby}
~~~
While the code-block renders as expected, inline-code is broken.
I have opened https://github.com/gettalong/kramdown/issues/752 to get kramdown author's opinion.

**Comment by ashmaroli** [comment]
This issue has been fixed with the release of `kramdown-2.4.0`. Please update at your end.
Closing..
